pkg/config/cos.go

```diff
@@ -325,49 +303,57 @@ func SaveOriginalNetworkConfig() error {
 	return err
 }
 
-// UpdateNetworkConfig updates a cOS config stage to include steps that:
+// UpdateManagementInterfaceConfig updates a cOS config stage to include steps that:
 // - generates wicked interface files (`/etc/sysconfig/network/ifcfg-*` and `ifroute-*`)
 // - manipulates nameservers in `/etc/resolv.conf`.
 // - call `wicked ifreload all` if `run` flag is true.
-func UpdateNetworkConfig(stage *yipSchema.Stage, networks map[string]Network, run bool) error {
-	mgmtNetwork, ok := networks[MgmtInterfaceName]
-	if !ok {
-		return errors.New("no management network defined")
+func UpdateManagementInterfaceConfig(stage *yipSchema.Stage, mgmtInterface Network, run bool) (string, error) {
+	if len(mgmtInterface.Interfaces) == 0 {
+		return "", errors.New("no slave defined for management network bond")
 	}
-	if len(mgmtNetwork.Interfaces) == 0 {
-		return errors.New("no slave defined for management network bond")
+
+	switch mgmtInterface.Method {
```

Comment:
What about using if expression rather than switch here because we do nothing in the case branch?

Reply:
we will need 3 conditions to match.
but switch-case could use 1 case to cover